packages/split-view/src/SplitView.ts

```diff
@@ -0,0 +1,416 @@
+        min: 's-resize',
+        max: 'n-resize',
+    },
+};
+
+const COLLAPSE_THREASHOLD = 50;
+
+const SPLITTERSIZE = 2;
+
+/**
+ * @element sp-split-view
+ */
+export class SplitView extends SpectrumElement {
+    public static get styles(): CSSResultArray {
+        return [
+            styles,
+            css`
+                .pane {
+                    overflow: hidden;
+                }
+                .secondaryPane {
+                    flex: 1;
+                }
+                #splitter {
+                    height: auto; /* for horizontal splitviews without proper outter height value */
+                }
+            `,
+        ];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public vertical = false;
+
+    @property({ type: Boolean, reflect: true })
+    public resizable = false;
+
+    @property({ type: Boolean, reflect: true })
+    public collapsible = false;
+
+    /** The minimum size of the primary pane */
+    @property({ type: Number, attribute: 'primary-min' })
+    public primaryMin = 304;
+
+    /** The maximum size of the primary pane */
+    @property({ type: Number, attribute: 'primary-max' })
+    public primaryMax = Infinity;
+
+    /** The size of the primary pane */
+    @property({ type: Number, attribute: 'primary-size' })
+    public primarySize?: number;
+
+    /** The default size of the primary pane */
+    @property({ type: Number, attribute: 'primary-default' })
+    public primaryDefault = 304;
+
+    /** The minimum size of the secondary pane */
+    @property({ type: Number, attribute: 'secondary-min' })
+    public secondaryMin = 304;
+
+    /** The maximum size of the secondary pane */
+    @property({ type: Number, attribute: 'secondary-max' })
+    public secondaryMax = Infinity;
+
+    @property({ type: Number, attribute: false })
+    public dividerPosition = this.primaryDefault;
+
+    @property({ type: Number, attribute: false })
+    public minPos = 0;
+
+    @property({ type: Number, attribute: false })
+    public maxPos = Infinity;
+
+    @property({ type: Boolean, reflect: true, attribute: 'is-collapsed-start' })
+    public isCollapsedStart = this.dividerPosition === 0;
+
+    @property({ type: Boolean, reflect: true, attribute: 'is-collapsed-end' })
+    public isCollapsedEnd = false;
+
+    @property({ type: Boolean, reflect: true })
+    public dragging = false;
+
+    @property({ type: Boolean, reflect: true })
+    public hovered = false;
+
+    @property()
+    public label?: string;
+
+    private offset = 0;
+    private size = 0;
+    private isOver = false;
+    private lastPosition = 0;
+
+    public constructor() {
+        super();
+        this.onPointerDragged = this.onPointerDragged.bind(this);
+        this.onPointerUp = this.onPointerUp.bind(this);
+        this.resize = this.resize.bind(this);
+        this.updatePosition = this.updatePosition.bind(this);
+        this.getSplitterSize = this.getSplitterSize.bind(this);
```

Comment:
If we're able to remove the `window.addEventListener()` code based on notes below, I'd like to not be binding these event like this any more.

Reply:
Removed all window.addEventListener() completely.